src/App.js

```diff
@@ -63,25 +60,76 @@ const App = () => {
-                                        element={<CreateBook 
-                                                    urlButton={urlButton}
-                                                    setUrlButton={setUrlButton}
-                                                />} />
-                                <Route path="/books/edit/:bookId" element={<CreateBook />} />
-                                <Route path="/books/:bookId" element={<SingleBook />} />
-                                //Using one component to show all user's conversations and another for a single conversation
-                                <Route path="/chat/:recipientId" element={<Chat/>} />
-                                <Route path="/chat/" element={<AllConversations/>} />
-                                <Route path="/my-profile" element={<ProfilePage />} />
+                                <Route
+                                    path="/search"
+                                    element={
+                                        <ProtectedRoute requiredAuthLevel="user">
+                                            <SearchPage />
+                                        </ProtectedRoute>
+                                    }
+                                />
+                                <Route
+                                    path="/books/create"
+                                    element={
+                                        <ProtectedRoute requiredAuthLevel="user">
+                                            <CreateBook />
+                                        </ProtectedRoute>
+                                    }
+                                />
+                                <Route
+                                    path="/books/edit/:bookId"
+                                    element={
+                                        <ProtectedRoute requiredAuthLevel="user">
+                                            <CreateBook />
+                                        </ProtectedRoute>
+                                    }
+                                />
+                                <Route
+                                    path="/books/:bookId"
+                                    element={
+                                        <ProtectedRoute requiredAuthLevel="user">
+                                            <SingleBook />
+                                        </ProtectedRoute>
+                                    }
+                                />
+                                <Route
+                                    path="/my-profile"
+                                    element={
+                                        <ProtectedRoute requiredAuthLevel="user">
+                                            <ProfilePage />
+                                        </ProtectedRoute>
+                                    }
+                                />
                             </Routes>
```

Comment:
Could you please add Protected route for chat routes that were accidentally removed? Also could you please wrap HomePage with Protected route?

Reply:
We had made a conscious decision to leave HomePage unprotected. Let's double check with Jacob/Nicki regarding that change.